aptos-move/framework/aptos-stdlib/sources/data_structures/ordered_map.move

```diff
@@ -0,0 +1,786 @@
+
+        if (index < len && &self.entries[index].key == &key) {
+            let Entry {
+                key: _,
+                value: old_value,
+            } = self.entries.replace(index, Entry { key, value });
+            option::some(old_value)
+        } else {
+            self.entries.insert(index, Entry { key, value });
+            option::none()
+        }
+    }
+
+    /// Remove a key/value pair from the map.
+    /// Aborts with EKEY_NOT_FOUND if `key` doesn't exist.
+    public fun remove<K: drop, V>(self: &mut OrderedMap<K, V>, key: &K): V {
+        let len = self.entries.length();
+        let index = binary_search(key, &self.entries, 0, len);
+        assert!(index < len, error::invalid_argument(EKEY_NOT_FOUND));
+        let Entry { key: old_key, value } = self.entries.remove(index);
+        assert!(key == &old_key, error::invalid_argument(EKEY_NOT_FOUND));
+        value
+    }
+
+    /// Returns whether map contains a given key.
+    public fun contains<K, V>(self: &OrderedMap<K, V>, key: &K): bool {
+        !self.find(key).iter_is_end(self)
+    }
+
+    public fun borrow<K, V>(self: &OrderedMap<K, V>, key: &K): &V {
+        self.find(key).iter_borrow(self)
+    }
+
+    public fun borrow_mut<K, V>(self: &mut OrderedMap<K, V>, key: &K): &mut V {
+        self.find(key).iter_borrow_mut(self)
+    }
+
+    /// Changes the key, with keeping the same value attached to it
+    /// Aborts with EKEY_NOT_FOUND if `old_key` doesn't exist.
+    /// Aborts with ENEW_KEY_NOT_IN_ORDER if `new_key` doesn't keep the order `old_key` was in.
+    public(friend) fun replace_key_inplace<K: drop, V>(self: &mut OrderedMap<K, V>, old_key: &K, new_key: K) {
+        let len = self.entries.length();
+        let index = binary_search(old_key, &self.entries, 0, len);
+        assert!(index < len, error::invalid_argument(EKEY_NOT_FOUND));
+
+        if (index > 0) {
+            assert!(cmp::compare(&self.entries[index - 1].key, &new_key).is_lt(), error::invalid_argument(ENEW_KEY_NOT_IN_ORDER))
+        };
+
+        if (index + 1 < len) {
+            assert!(cmp::compare(&new_key, &self.entries[index + 1].key).is_lt(), error::invalid_argument(ENEW_KEY_NOT_IN_ORDER))
+        };
+
+        let entry = self.entries.borrow_mut(index);
+        assert!(old_key == &entry.key, error::invalid_argument(EKEY_NOT_FOUND));
+        entry.key = new_key;
+    }
+
+    /// Add multiple key/value pairs to the map. The keys must not already exist.
+    /// Aborts with EKEY_ALREADY_EXISTS if key already exist, or duplicate keys are passed in.
+    public fun add_all<K, V>(self: &mut OrderedMap<K, V>, keys: vector<K>, values: vector<V>) {
+        vector::zip(keys, values, |key, value| {
+            add(self, key, value);
+        });
+    }
+
+    /// Takes all elements from `other` and adds them to `self`.
+    /// Aborts with EKEY_ALREADY_EXISTS if `other` has a key already present in `self`.
```

Comment:
append_unique or append_disjoint? I think overwriting should be the default (as in rust btreemap, for example)
 no big deal since it's well documented.